### PR TITLE
Update Java SDK docs to 3.0

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
Documentation and guides for Java SDK 3.0

### CODE REVIEW

File: src/platforms/java/common/legacy/google-app-engine/index.mdx [hunk 1]
```diff
@@ -1,9 +1,13 @@
 ---
 title: Google App Engine
-redirect_from:
-  - /clients/java/modules/appengine/
+sidebar_order: 30
 ---
 
+<Alert level="warning" title="Note"><markdown>
+
+A new Java SDK has superseded this deprecated version. Sentry preserves this documentation for customers using the old client. We recommend using the [updated Java SDK](/platforms/java/) for new projects.
```
Comment (PeloWriter) [inline src/platforms/java/common/legacy/google-app-engine/index.mdx:8]:
```suggestion
An [updated Java SDK](/platforms/java/) Java SDK has superseded this deprecated version. Sentry preserves this documentation for customers using the old client. We recommend using the updated Java SDK for new projects.
```

File: src/platforms/java/common/legacy/index.mdx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+title: Legacy SDK (1.7)
+sidebar_order: 2000
+---
+
+This section covers legacy Sentry SDK for Java (1.7.x). Legacy SDK is not under active development anymore, and we **strongly encourage** updating to the most recent Sentry SDK.
```
Comment (PeloWriter) [inline src/platforms/java/common/legacy/index.mdx:6]:
```suggestion
This section covers the legacy Sentry SDK for Java (1.7.x), which is no longer under active development. We **strongly encourage** updating to [the updated Java SDK](/platforms/java/).
```

File: src/platforms/java/common/legacy/log4j/index.mdx [hunk 3]
```diff
@@ -1,9 +1,13 @@
 ---
 title: log4j 1.x
-redirect_from:
-  - /clients/java/modules/log4j/
+sidebar_order: 30
 ---
 
+<Alert level="warning" title="Note"><markdown>
+
+A new Java SDK has superseded this deprecated version. Sentry preserves this documentation for customers using the old client. We recommend using the [updated Java SDK](/platforms/java/) for new projects.
```
Comment (PeloWriter) [inline src/platforms/java/common/legacy/log4j/index.mdx:8]:
```suggestion
An [updated Java SDK](/platforms/java/) supersedes this deprecated version. Sentry preserves this documentation for customers using the old client. We recommend using the updated Java SDK for new projects.
```

File: src/platforms/java/common/legacy/log4j2/index.mdx [hunk 4]
```diff
@@ -0,0 +1,140 @@
+---
+title: log4j 2.x
+sidebar_order: 30
+---
+
+<Alert level="warning" title="Note"><markdown>
+
+A new Java SDK has superseded this deprecated version. Sentry preserves this documentation for customers using the old client. We recommend using the [updated Java SDK](/platforms/java/) for new projects.
```
Comment (PeloWriter) [inline src/platforms/java/common/legacy/log4j2/index.mdx:8]:
```suggestion
An [updated Java SDK](/platforms/java/) SDK supersedes this deprecated version. Sentry preserves this documentation for customers using the old client. We recommend using the updated Java SDK for new projects.
```

File: src/platforms/java/common/legacy/logging/index.mdx [hunk 5]
```diff
@@ -1,10 +1,14 @@
...
 title: java.util.logging
-redirect_from:
-  - /clients/java/modules/jul/
+sidebar_order: 30
 ---
 
-The `sentry` library provides a [java.util.logging Handler](https://docs.oracle.com/javase/7/docs/api/java/util/logging/Handler.html) that sends logged exceptions to Sentry. Once this integration is configured you can _also_ use Sentry’s static API, [as shown on the usage page](usage/), in order to do things like record breadcrumbs, set the current user, or manually send events.
+<Alert level="warning" title="Note"><markdown>
+
+A new Java SDK has superseded this deprecated version. Sentry preserves this documentation for customers using the old client. We recommend using the [updated Java SDK](/platforms/java/) for new projects.
```
Comment (PeloWriter) [inline src/platforms/java/common/legacy/logging/index.mdx:8]:
```suggestion
An [updated Java SDK](/platforms/java/) supersedes this deprecated version. Sentry preserves this documentation for customers using the old client. We recommend using the updated Java SDK for new projects.
```

File: src/platforms/java/common/legacy/spring/index.mdx [hunk 6]
```diff
@@ -0,0 +1,79 @@
+---
+title: Spring
+sidebar_order: 30
+---
+
+<Alert level="warning" title="Note"><markdown>
+
+A new Java SDK has superseded this deprecated version. Sentry preserves this documentation for customers using the old client. We recommend using the [updated Java SDK](/platforms/java/) for new projects.
```
Comment (PeloWriter) [inline src/platforms/java/common/legacy/spring/index.mdx:8]:
```suggestion
An [updated Java SDK](/platforms/java/) supersedes this deprecated version. Sentry preserves this documentation for customers using the old client. We recommend using the updated Java SDK for new projects.
```

File: src/platforms/java/guides/log4j2/index.mdx [hunk 7]
```diff
@@ -14,123 +14,214 @@ The source can be found [on GitHub](https://github.com/getsentry/sentry-java/tre
...
+Following example shows how to configure DSN from `SENTRY_DSN` the environment variable.
 
-### Additional Data
+```xml
+<Sentry name="Sentry"
+        dsn="${env:SENTRY_DSN}" />
+```
 
-It’s possible to add extra data to events thanks to [the marker system](https://logging.apache.org/log4j/2.x/manual/markers.html) provided by Log4j 2.x.
+### Minimum log level
```
Comment (PeloWriter) [inline src/platforms/java/guides/log4j2/index.mdx:83]:
```suggestion
### Minimum Log Level
```

File: src/platforms/java/guides/spring-boot/index.mdx [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+        val user = User()
+        // ... set user information
+        return user
+    }
+}
+```
+
+#### Using Git Commit ID As The Release
+
+When Spring Boot is [configured to generate Git information](https://docs.spring.io/spring-boot/docs/current/reference/html/howto.html#howto-git-info) every event triggered by Sentry will have a `release` field set to the current Git commit ID that will enable [Monitor Release Health](https://docs.sentry.io/product/releases/health/) Sentry feature.
```
Comment (PeloWriter) [inline src/platforms/java/guides/spring-boot/index.mdx:104]:
```suggestion
When Spring Boot is [configured to generate Git information](https://docs.spring.io/spring-boot/docs/current/reference/html/howto.html#howto-git-info) every event triggered by Sentry will have a `release` field set to the current Git commit ID that will enable [Monitor Release Health](/product/releases/health/) Sentry feature.
```

**Comment by maciejwalkowiak** [comment]
Thanks @PeloWriter for first review round!

Tomorrow I'll add a guide for Servlet integration.

File: src/includes/configuration/config-intro/java.mdx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+```java {tabTitle:Java}
+import io.sentry.Sentry;
+
+public class MyClass {
+  public static void main(String... args) {
+    Sentry.init(
+        options -> {
+          // NOTE: Replace the test DSN below with YOUR OWN DSN to see the events from this app in
+          // your Sentry project/dashboard
+          options.setDsn("https://f7f320d5c3a54709be7b28e0f2ca7081@sentry.io/1808954");
```
Comment (bruno-garcia) [inline src/includes/configuration/config-intro/java.mdx:13]:
```suggestion
          // your Sentry project/dashboard
          options.setDsn("___PUBLIC_DSN___");
```

File: src/includes/configuration/config-intro/java.mdx [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+```kotlin {tabTitle:Kotlin}
+import io.sentry.SentryOptions.BeforeBreadcrumbCallback
+import io.sentry.SentryOptions.BeforeSendCallback
+
+fun main() {
+    Sentry.init {
+        // NOTE: Replace the test DSN below with YOUR OWN DSN to see the events from this app in
+        // your Sentry project/dashboard
+        it.dsn = "https://f7f320d5c3a54709be7b28e0f2ca7081@sentry.io/1808954"
```
Comment (bruno-garcia) [inline src/includes/configuration/config-intro/java.mdx:36]:
```suggestion
        // your Sentry project/dashboard
        it.dsn = "___PUBLIC_DSN___"
```

File: src/platforms/java/guides/log4j2/index.mdx [hunk 11]
```diff
@@ -14,123 +14,200 @@ The source can be found [on GitHub](https://github.com/getsentry/sentry-java/tre
...
-    <appenders>
+<Configuration status="warn" packages="org.apache.logging.log4j.core,io.sentry.log4j2">
+    <Appenders>
         <Console name="Console" target="SYSTEM_OUT">
-            <PatternLayout pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n" />
+            <PatternLayout pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
         </Console>
+        <!-- NOTE: Replace the test DSN below with YOUR OWN DSN to see the events from this app in your Sentry project/dashboard -->
+        <Sentry name="Sentry"
+                dsn="https://9eda1857f9344d51821b656ba3557780@o420886.ingest.sentry.io/5339853" />
```
Comment (bruno-garcia) [inline src/platforms/java/guides/log4j2/index.mdx:46]:
```suggestion
        <Sentry name="Sentry"
                dsn="___PUBLIC_DSN___" />
```

File: src/platforms/java/guides/log4j2/index.mdx [hunk 12]
```diff
@@ -14,123 +14,200 @@ The source can be found [on GitHub](https://github.com/getsentry/sentry-java/tre
...
-            <appender-ref ref="Sentry" level="WARN" />
-        </root>
-    </loggers>
-</configuration>
+Note that **you need to configure your DSN** (client key).
+
+```xml
+<!-- NOTE: Replace the test DSN below with YOUR OWN DSN to see the events from this app in your Sentry project/dashboard -->
+<Sentry name="Sentry"
+        dsn="https://9eda1857f9344d51821b656ba3557780@o420886.ingest.sentry.io/5339853" />
```
Comment (bruno-garcia) [inline src/platforms/java/guides/log4j2/index.mdx:64]:
```suggestion
<Sentry name="Sentry"
        dsn="___PUBLIC_DSN___" />
```

File: src/platforms/java/guides/log4j2/index.mdx [hunk 13]
```diff
@@ -14,123 +14,200 @@ The source can be found [on GitHub](https://github.com/getsentry/sentry-java/tre
...
 
-It’s possible to add extra data to events thanks to [the marker system](https://logging.apache.org/log4j/2.x/manual/markers.html) provided by Log4j 2.x.
+```xml
+<!-- NOTE: Replace the test DSN below with YOUR OWN DSN to see the events from this app in your Sentry project/dashboard -->
+<!-- Setting minimumBreadcrumbLevel modifies the default minimum level to add breadcrumbs from INFO to DEBUG  -->
+<!-- Setting minimumEventLevel the default minimum level to capture an event from ERROR to WARN  -->
+<Sentry name="Sentry"
+        dsn="https://9eda1857f9344d51821b656ba3557780@o420886.ingest.sentry.io/5339853"
+        minimumBreadcrumbLevel="DEBUG"
```
Comment (bruno-garcia) [inline src/platforms/java/guides/log4j2/index.mdx:79]:
```suggestion
<!-- Setting minimumBreadcrumbLevel modifies the default minimum level to add breadcrumbs from INFO to DEBUG  -->
<!-- Setting minimumEventLevel the default minimum level to capture an event from ERROR to WARN  -->
<Sentry name="Sentry"
        dsn="___PUBLIC_DSN___"
        minimumBreadcrumbLevel="DEBUG"
```

**Review by bruno-garcia — COMMENTED** [review]
use `___PUBLIC_DSN___` to get replaced by logged in users' dsn

File: src/platforms/java/common/scope.mdx [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+import io.sentry.protocol.User;
+
+public class MyClass {
+
+  public void example() {
+    // Manually initialize the static client, you must also pass in a DSN and/or
+    // SentryClientFactory to use. Note that the client will attempt to automatically
+    // initialize on the first use of the static API, so this isn't strictly necessary.
+    Sentry.init(options -> {
+      options.setDsn("https://f7f320d5c3a54709be7b28e0f2ca7081@sentry.io/1808954");
```
Comment (bruno-garcia) [inline src/platforms/java/common/scope.mdx:29]:
```suggestion
      options.setDsn("___PUBLIC_DSN___");
```

**Comment by bruno-garcia** [comment]
I suggest we delete the 'migrating' doc. Or move it under `legacy` and title it:
Migrating between `Raven` and version 1.7.
https://sentry-docs-git-fork-maciejwalkowiak-sentry-java-30.sentry.dev/platforms/java/migration/

**Comment by bruno-garcia** [comment]
Maybe worth hidding the generic breadcrumbs doc too: https://sentry-docs-git-fork-maciejwalkowiak-sentry-java-30.sentry.dev/platforms/java/enriching-events/breadcrumbs/

Since Java has its own. Right now 2 breadcrumbs items show on the list

**Review by bruno-garcia — APPROVED** [review]
Suggested some improvements.
But it LGTM.

We can merge, release the SDK and iterate improving things.

**Comment by maciejwalkowiak** [comment]
The problem with "Legacy" docs placement still remains. Since it's added to "Common" section, each guide has now a legacy section and you can end up on the site like `Home > Platforms > Java > Spring Boot > Legacy SDK (1.7) > log4j 2.x`

image

File: src/includes/configuration/config-intro/java.mdx [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+                }
+                return event;
+              });
+        });
+  }
+}
+```
+
+```kotlin {tabTitle:Kotlin}
+import io.sentry.SentryOptions.BeforeBreadcrumbCallback
```
Comment (marandaneto) [inline src/includes/configuration/config-intro/java.mdx:28]: do we need this? its not using it

File: src/includes/configuration/drain-example/java.mdx [hunk 16]
```diff
@@ -0,0 +1 @@
+The Java SDK automatically shuts down on JVM exit and waits `shutdownTimeout` seconds before that happens.
```
Comment (marandaneto) [inline src/includes/configuration/drain-example/java.mdx:1]:
```suggestion
The Java SDK automatically shuts down on JVM exit and waits `shutdownTimeout` milliseconds before that happens.
```

File: src/platforms/java/common/legacy/configuration/index.mdx [hunk 17]
```diff
@@ -0,0 +1,467 @@
...
+Sentry **will not** be able to do anything with events until this line is run, so this method of configuration is not recommended if you might have errors occur during startup. In addition, by passing a hardcoded DSN you are no longer able to override the DSN at runtime via Java System Properties or System Environment Variables.
+
+### Configuration via the DSN
+
+The SDK can also be configured by setting querystring parameters on the DSN itself. This is a bit recursive because your DSN itself is an option that you must set somewhere (and not in the DSN!).
+
+Option names in the DSN exactly match the examples given below. For example, to enable sampling if you are setting your DSN via the environment:
+
+```bash
+SENTRY_DSN={DSN}/1?sample.rate=0.75 java -jar app.jar
```
Comment (marandaneto) [inline src/platforms/java/common/legacy/configuration/index.mdx:95]:
```suggestion
SENTRY_DSN=___PUBLIC_DSN___/1?sample.rate=0.75 java -jar app.jar
```

File: src/platforms/java/guides/logback/index.mdx [hunk 18]
```diff
@@ -64,63 +61,103 @@ Example configuration using the `logback.xml` format:
...
-Next, **you’ll need to configure your DSN** (client key) and optionally other values such as `environment` and `release`. [See the configuration page](configuration/options/#setting-the-dsn) for ways you can do this.
+### DSN Configuration
+
+Note that **you need to configure your DSN** (client key).
+
+```xml
+<appender name="Sentry" class="io.sentry.logback.SentryAppender">
+    <options>
+        <!-- NOTE: Replace the test DSN below with YOUR OWN DSN to see the events from this app in your Sentry project/dashboard -->
+        <dsn>https://f7f320d5c3a54709be7b28e0f2ca7081@sentry.io/1808954</dsn>
```
Comment (marandaneto) [inline src/platforms/java/guides/logback/index.mdx:72]:
```suggestion
        <dsn>___PUBLIC_DSN___</dsn>
```

File: src/platforms/java/guides/logback/index.mdx [hunk 19]
```diff
@@ -64,63 +61,103 @@ Example configuration using the `logback.xml` format:
...
+### Advanced Configuration
+
+<!-- TODO: link to correct site with explanation of each configuration property -->
+Optionally you can configure other values such as `environment` and `release`. [See the configuration page](configuration/options/#setting-the-dsn) for the in-depth explanation of each property.
+
+```xml
+<appender name="Sentry" class="io.sentry.logback.SentryAppender">
+    <options>
+        <!-- NOTE: Replace the test DSN below with YOUR OWN DSN to see the events from this app in your Sentry project/dashboard -->
+        <dsn>https://f7f320d5c3a54709be7b28e0f2ca7081@sentry.io/1808954</dsn>
```
Comment (marandaneto) [inline src/platforms/java/guides/logback/index.mdx:88]:
```suggestion
        <dsn>___PUBLIC_DSN___</dsn>
```

File: src/platforms/java/guides/logback/index.mdx [hunk 20]
```diff
@@ -64,63 +61,103 @@ Example configuration using the `logback.xml` format:
...
 
-<!-- TODO-ADD-VERIFICATION-EXAMPLE -->
+```xml
+<appender name="Sentry" class="io.sentry.logback.SentryAppender">
+    <options>
+        <!-- NOTE: Replace the test DSN below with YOUR OWN DSN to see the events from this app in your Sentry project/dashboard -->
+        <dsn>https://f7f320d5c3a54709be7b28e0f2ca7081@sentry.io/1808954</dsn>
```
Comment (marandaneto) [inline src/platforms/java/guides/logback/index.mdx:113]:
```suggestion
        <dsn>___PUBLIC_DSN___</dsn>
```

File: src/platforms/java/common/legacy/logging/index.mdx [hunk 21]
```diff
@@ -1,10 +1,14 @@
...
+sidebar_order: 30
 ---
 
-The `sentry` library provides a [java.util.logging Handler](https://docs.oracle.com/javase/7/docs/api/java/util/logging/Handler.html) that sends logged exceptions to Sentry. Once this integration is configured you can _also_ use Sentry’s static API, [as shown on the usage page](usage/), in order to do things like record breadcrumbs, set the current user, or manually send events.
+<Alert level="warning" title="Note"><markdown>
+
+An [updated Java SDK](/platforms/java/) supersedes this deprecated version. Sentry preserves this documentation for customers using the old client. We recommend using the updated Java SDK for new projects.
+</markdown></Alert>
+
+The `sentry` library provides a [java.util.logging Handler](http://docs.oracle.com/javase/7/docs/api/java/util/logging/Handler.html) that sends logged exceptions to Sentry. Once this integration is configured you can _also_ use Sentry’s static API, [as shown on the usage page](/platforms/java/legacy/usage), in order to do things like record breadcrumbs, set the current user, or manually send events.
```
Comment (marandaneto) [inline src/platforms/java/common/legacy/logging/index.mdx:11]:
```suggestion
The `sentry` library provides a [java.util.logging Handler](http://docs.oracle.com/javase/8/docs/api/java/util/logging/Handler.html) that sends logged exceptions to Sentry. Once this integration is configured you can _also_ use Sentry’s static API, [as shown on the usage page](/platforms/java/legacy/usage), in order to do things like record breadcrumbs, set the current user, or manually send events.
```

File: src/platforms/java/common/legacy/configuration/index.mdx [hunk 22]
```diff
@@ -0,0 +1,467 @@
...
+
+It’s possible to manually set the number of threads (for example if you want only one thread) with the option `async.threads`:
+
+```
+async.threads=1
+```
+
+#### Async Threads Priority (Advanced)
+
+In most cases sending logs to Sentry isn’t as important as an application running smoothly, so the threads have a [minimal priority](http://docs.oracle.com/javase/6/docs/api/java/lang/Thread.html#MIN_PRIORITY).
```
Comment (marandaneto) [inline src/platforms/java/common/legacy/configuration/index.mdx:339]:
```suggestion
In most cases sending logs to Sentry isn’t as important as an application running smoothly, so the threads have a [minimal priority](http://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#MIN_PRIORITY).
```

File: src/includes/configuration/config-intro/java.mdx (hunk 15 above)
Comment (maciejwalkowiak) [reply src/includes/configuration/config-intro/java.mdx:28]: 👍

**Comment by bruno-garcia** [comment]
Let's merge and iterate over this?!

File: src/platforms/java/common/legacy/configuration/index.mdx (hunk 22 above)
Comment (bruno-garcia) [reply src/platforms/java/common/legacy/configuration/index.mdx:339]: that's legacy stuff

Comment (marandaneto) [reply src/platforms/java/common/legacy/configuration/index.mdx:339]: ups

**Pull request merged**